solver/src/settlement/settlement_encoder.rs

```diff
@@ -548,7 +548,7 @@ pub mod tests {
     fn merge_fails_because_price_is_different() {
         let prices = hashmap! { token(1) => 1.into(), token(2) => 2.into() };
         let encoder0 = SettlementEncoder::new(prices);
-        let prices = hashmap! { token(1) => 1.into(), token(2) => 3.into() };
+        let prices = hashmap! { token(1) => 1.into(), token(2) => 4.into() };
```

Comment:
What does this change have to do with the purpose of this PR?

Reply:
It's fixing a flaky unit test. Depending on the order we iterate over the hashset we may scale token 1 (scale factor 1, test works as expected) or scale token 2 (scale factor 3/2 which when multiplied with 1 results in 1, so the merge doesn't fail as expected).

Reply:
I see now why this was changed here! I hope we can land this PR soon because of this.

Reply:
Feel free to copy the change if this is blocking another PR